### PR TITLE
Dave prepop eeitt 478

### PR DESCRIPTION
**Pull request by lutnos**
Data storage in save4later for DFS prepop data

### CODE REVIEW

File: test/uk/gov/hmrc/eeitt/controllers/PrepopulationDataControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      MicroserviceShortLivedCache.cache[JsValue](cacheId, formId, body)
+
+    def remove(cacheId: String): Future[HttpResponse] =
+      MicroserviceShortLivedCache.remove(cacheId)(hc)
+
+  }
+
+  "GET /prepopulation/:cacheId/:formId" should {
+    "return 404 for unknown ids" in {
+      await(PrepopulationDataHelper.remove("i-and-and"))
```
Comment (VlachJosef) [inline test/uk/gov/hmrc/eeitt/controllers/PrepopulationDataControllerSpec.scala]: I'm getting 
```
uk.gov.hmrc.play.http.BadGatewayException: DELETE of 'http://localhost:9272/save4later/eeitt/i-and-and' failed. Caused by: 'Connection refused: localhost/0:0:0:0:0:0:0:1:9272'
```
when running this test locally.

It should be moved into integration test scope if it depends on live system.

File: app/uk/gov/hmrc/eeitt/controllers/RegistrationController.scala [hunk 2]
```diff
@@ -51,6 +51,7 @@ class RegistrationController(
 trait RegistrationControllerHelper extends BaseController with I18nSupport {
 
   def verify[A: FindRegistration](findParams: A) = Action.async { implicit request =>
+    Logger.info("We here ZZZZZZZZZZZ")
```
Comment (lutnos) [inline app/uk/gov/hmrc/eeitt/controllers/RegistrationController.scala]: Oops, thanks

File: test/uk/gov/hmrc/eeitt/controllers/PrepopulationDataControllerSpec.scala (hunk 1 above)
Comment (lutnos) [inline test/uk/gov/hmrc/eeitt/controllers/PrepopulationDataControllerSpec.scala]: Yes, will do

File: app/uk/gov/hmrc/eeitt/controllers/PrepopulationDataController.scala [hunk 3]
```diff
@@ -8,32 +8,28 @@ import uk.gov.hmrc.play.http.HttpResponse
...
-          case c: CacheMap =>
-            Ok
-        }
-    }.getOrElse(Future.successful(BadRequest))
-  }
+    def put(cacheId: String, formId: String) = Action.async(parse.json) { implicit request =>
+      MicroserviceShortLivedCache.cache[JsValue](cacheId, formId, request.body).map {
+        case c: CacheMap =>
+          Ok
+        case _ =>
```
Comment (VlachJosef) [inline app/uk/gov/hmrc/eeitt/controllers/PrepopulationDataController.scala]: Why this case is here `MicroserviceShortLivedCache.cache` is returning `Future[CacheMap]` so simple `.map(_ => Ok)` should be enough?
Maybe you want `recover` from failed future?

**Pull request merged**